### PR TITLE
[sso] split corehq.app.sso.utils into a module and have better naming for functionality

### PR DESCRIPTION
**Pull request by biyeun**
## Summary
This is a response to https://github.com/dimagi/commcare-hq/pull/29180#discussion_r578516083

This PR creates a `sso.utils` module and splits up the methods in `utils.py` into their own separate functional files with more appropriate/clearer naming

I also took the opportunity to rename a poorly/vaguely named `classmethod` of `IdentityProvider`.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Any of the forms or decorator SSO tests will catch issues if any :)

### QA Plan
No QA needed.

### Safety story
Does not affect active code. Just a simple code refactor.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/sso/models.py [hunk 1]
```diff
@@ -128,11 +128,6 @@ def get_sso_exempt_users(self):
             email_domain__identity_provider=self,
         ).values_list('username', flat=True)
 
-    @classmethod
-    def domain_has_identity_provider(cls, domain):
-        owner = BillingAccount.get_account_by_domain(domain)
-        return cls.objects.filter(owner=owner, is_editable=True).exists()
-
```
Comment (mjriley) [inline corehq/apps/sso/models.py:135]: Interesting -- I probably would have left this as is. I've been trying to move any `objects` access to within the associated model class, to make mocking and reading easier. What pushed you to move this to its own module?

**Review by mjriley — APPROVED** [review]
Looks good

File: corehq/apps/sso/models.py (hunk 1 above)
Comment (biyeun) [reply corehq/apps/sso/models.py:135]: i guess renaming it would make more sense...hmm. on second thought I'm going to move this back

Comment (biyeun) [reply corehq/apps/sso/models.py:135]: what it really needs is a rename

**Pull request merged**